pkg/cvo/sync_worker.go

```diff
@@ -499,9 +499,7 @@ func (w *SyncWorker) Update(ctx context.Context, generation int64, desired confi
 		}
 	}
 
-	w.lock.Unlock()
 	implicit, err := w.loadUpdatedPayload(ctx, work, cvoOptrName)
```

Comment:
There's another unlock/lock around a different `loadUpdatedPayload` ~30 lines up^^ that you probably want to drop too.